addons/mrp/tests/test_stock.py

```diff
@@ -316,6 +316,42 @@ def test_backorder_unpacking(self):
         self.assertEqual(backorder.state, 'done')
         self.assertEqual(mo.move_raw_ids.move_line_ids.mapped('quantity_product_uom'), [20, 80])
 
+    def test_active_rules_visibility_in_routes(self):
+        """
+            Test to ensure that only those rules are shown in routes which are active itself and it's destination locations is active.
+            if both are inactive then rules is not shown in routes.
+        """
+
```

Comment:
- For the name of the test, I'm not sure my suggestion is really good but I would avoid to speak about visibility because it's not the issue;
- I suggest another description because the current one can be misleading ("*if both are inactive then rules is not shown in routes*" let think that prior to the fix, a rule can be shown even if inactive which is not the case) and to explain the tested flow 
```suggestion
    def test_unarchive_mto_route_active_needed_rules_only(self):
        """ Ensure that activating a route will activate only its relevant rules.
            Here, unarchiving the MTO route shouldn't active pull rule for the Pre-Production
            location if manufacture is in 1 step since this location is archived.
        """
```